### DISCUSSION TITLE
Not able to deploy in zksync sepholia testnet.

### DISCUSSION THREAD

**Discussion by coder-002** [Cyfrin/foundry-full-course-cu#2245] [General]
I am new to solidity and am trying to deploy the contract. I followed all the instruction given in the video but also I am not able to deploy it 

![image](https://github.com/user-attachments/assets/bb4801ca-8b0f-4b1b-8324-c1e955b145c7)

**Comment by EngrPips** [comment]
Hello @coder-002, It seems as if you are using the `deploy and verify` option, but that doesn't work appropriately as of now, so uncheck the verify box and only deploy the contract. Doing so, things should work correctly.

**Reply by coder-002** [reply]
tried with checked as well as unchecked. In both condition no deployment is failed

**Reply by EngrPips** [reply]
Maybe refresh your browser then retry with `deployment` only, don't verify.

**Reply by Danthesage01** [reply]
Deploying on Zysknc Sepolia Testnet is still not working at my end. @coder-002 has it worked for you?

**Reply by EngrPips** [reply]
Are using `deploy and verify` or `deploy`?

**Reply by Danthesage01** [reply]
I tried both and even refreshed my browser, as you suggested above. It didn't work.

**Reply by EngrPips** [reply]
That's a tough one. Working with Zksync has been tough. I tried deploying to `ZkSync sepolia` now, and it was unsuccessful, but it easily deployed to the `Remote Devnet.` Can you try deploying to `Remote Devnet`? Let's see if that works for you as well

**Reply by Danthesage01** [reply]
Yes. When I tried it on Remote Devnet, it was successful, and I was able to interact with the contract. Only the deployment on the Zksync sepolia testnet with my wallet was unsuccessful.

**Reply by EngrPips** [reply]
Interesting. Seems like this is a problem with the plug-in itself. @PatrickAlphaC  we are having a hard time deploying to `ZkSync Sepolia` testnet, we would appreciate any help at all sir.

**Reply by PatrickAlphaC** [reply]
What's the error you're getting?

**Reply by EngrPips** [reply]
No error, sir, when I click the `deploy` button, it prompts my metamask wallet, and I sign the transaction, but then it just showed deployment failed after I signed the transaction and nothing else 

<img width="1440" alt="Screenshot 2024-08-19 at 16 00 23" src="https://github.com/user-attachments/assets/30c31268-c3ad-48eb-9999-8c9afeb741df">

**Reply by KevinFalah** [reply]
<img width="907" alt="image" src="https://github.com/user-attachments/assets/1bb9d4ed-454d-4dea-9502-766b6c18b96c">

I had a similar issue, when the metamask pop up appeared and click sign button, the popup just gone and deploy will be failed, any help and info is greatly appreciated sir @PatrickAlphaC

**Reply by DCVglobalnetwork** [reply]
Double-check your code It went through , cleared up and start over again refresh remix and start over again 

![image](https://github.com/user-attachments/assets/378f918b-8dbc-4af1-9178-18cbe1b0100b)


![image](https://github.com/user-attachments/assets/a331bc38-1d1b-4b1b-b6af-758e49ef3db0)

**Reply by DCVglobalnetwork** [reply]
Yes I was able to interact with the `FundMe` on zkSync testnet and send 0.01 ETH 

![image](https://github.com/user-attachments/assets/76ea77cb-aae1-42a3-89dd-6f1a81508388)


![image](https://github.com/user-attachments/assets/6ebda6ee-5757-4fd1-b792-b1a593a901c5)

**Comment by DCVglobalnetwork** [comment]
Hi, I had a similar issue, but than I did on different contracts and worked perfectly, I was able to deploy without error or waiting for ages. I did it, and I got an error. Can you suggest something with a big number? Is this related to gas? Anyone who can assist? Once again, this is only error I get with this specific contract

![image](https://github.com/user-attachments/assets/cdaa04dd-894d-4e9a-b23b-b5c2e4a8e5ca)

**Reply by EngrPips** [reply]
Can you uncheck the `verify` box and retry?

**Reply by DCVglobalnetwork** [reply]
yes sir I did without verify and failed

![image](https://github.com/user-attachments/assets/b5335769-3601-416d-a5af-acc6bd251ac4)

I also deployed on remote and went through 

![image](https://github.com/user-attachments/assets/42e94dc9-6cdb-4918-8aa9-7c150facb29f)

**Reply by EngrPips** [reply]
Interesting. This is an issue with deployment to `ZkSync Sepolia.` Let's all look forward to Patrick's comment on this.

**Reply by DCVglobalnetwork** [reply]
Ok, sir, but I was able to deploy other contracts and even interact with them on testnet, so if the plugin is not working well, could it be on every contract? 

![image](https://github.com/user-attachments/assets/72be812c-2e85-4f4e-85fd-44dd94029da1)

![image](https://github.com/user-attachments/assets/bbf67805-4320-4a72-9c27-790209424c08)

**Reply by EngrPips** [reply]
Hmmm, could it be a problem with the `SimpleStorage` contract?

**Reply by DCVglobalnetwork** [reply]
yes sir with the `SimpleStorage` contract, here is the error 

```sh 
You are using the latest version of the zkSync plugin: v5.0.0-0
Deploying contract SimpleStorage
Error: "transaction failed [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (transactionHash=\"0xf3578e3f8cb6de8dc8be69605f899845b46ea11c3158ddd4723ce89c8b9f5009\", transaction={\"type\":113,\"nonce\":7,\"maxPriorityFeePerGas\":{\"type\":\"BigNumber\",\"hex\":\"0x017d7840\"},\"maxFeePerGas\":{\"type\":\"BigNumber\",\"hex\":\"0x017d7840\"},\"gasLimit\":{\"type\":\"BigNumber\",\"hex\":\"0x048b3c\"},\"to\":\"0x0000000000000000000000000000000000008006\",\"value\":{\"type\":\"BigNumber\",\"hex\":\"0x00\"},\"data\":\"0x9c4d535b0000000000000000000000000000000000000000000000000000000000000000010000a19866d31c8d9f10ad6291eb7e9eaa7f2e3ed08ad4acb14066ab61316100000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000\",\"chainId\":{\"type\":\"BigNumber\",\"hex\":\"0x012c\"},\"from\":\"0x230138Bfb9547A0d06594a121a11b167Ab728CeF\",\"customData\":{\"gasPerPubdata\":{\"type\":\"BigNumber\",\"hex\":\"0xc350\"},\"factoryDeps\":
```
due to a failure in the transaction execution?

**Reply by EngrPips** [reply]
Patrick has been informed about this issue. Let's wait to hear from him. I am having issues trying to deploy myself, and I don't even know how to resolve it yet

**Reply by DCVglobalnetwork** [reply]
Sir I did it and it went through 

![image](https://github.com/user-attachments/assets/2d3f80d4-c687-4d7b-8826-a4381ced85a2)

![image](https://github.com/user-attachments/assets/781370aa-2a97-4362-a003-5639f40cf109)

![image](https://github.com/user-attachments/assets/aa61d722-0198-4cfc-b335-26589b4467d0)

**Reply by DCVglobalnetwork** [reply]
still sending 

![image](https://github.com/user-attachments/assets/4f782637-8038-4f83-9a64-eba1a1a4e500)

**Reply by EngrPips** [reply]
It seems to work for me too now.

<img width="1440" alt="Screenshot 2024-08-19 at 19 37 34" src="https://github.com/user-attachments/assets/be403b81-3e09-4438-9bd8-1a04e93b225c">

**Reply by Danthesage01** [reply]
Mine works now, too. Thank you, @EngrPips  and everyone

**Reply by EngrPips** [reply]
You are welcome.

**Reply by nbanguiano** [reply]
Hi, @EngrPips it's failing for me now. I see the last comment saying it worked was 10 hours ago. Maybe it's just down right now?

I can deploy to Dev Testnet, just not to the wallet. Getting the exact same (logless) error than coder-002 posted above.

If it's deploying to the test net, I'll take as a plugin error and not a code error, right?

Say yes, so I can sleep today!

**Reply by nbanguiano** [reply]
By the way, I was not able to deploy to Metamask via "injected provider" either.
<img width="1097" alt="Remix_-_Ethereum_IDE" src="https://github.com/user-attachments/assets/dcf82208-5d85-4841-8593-f9028927c15a">

Getting:
Gas estimation failed
Gas estimation errored with the following message (see below). The transaction execution will likely fail. Do you want to force sending?
Returned error: Failed to serialize transaction: toAddressIsNull
[object Object]

¯\_(ツ)_/¯

**Reply by EngrPips** [reply]
Hello @nbanguiano, Are you sure there is no logic error in your smart contract because this `Gas estimation` error will normally occur due to a contract logic error, also you can try the `wallet connect` option to connect your metamask to Remix if you are having issues with `injected Provider`

**Reply by DCVglobalnetwork** [reply]
Yes, as I stated below, the cause of the gas fee could be an issue with the test network.

**Reply by nbanguiano** [reply]
ahh, ok both issues were solved.

Not being able to deploy to Metamask via the zkSync plugin was just a matter of time. Today it worked.

The Gas estimation failed error was because I still had Metamast set to zkSync testnet (from when I did the bridge). So I changed it to Sepolia and everything worked!

Thanks @EngrPips & @DCVglobalnetwork

**Comment by C0dewithLokesh** [comment]
Hi everyone,

I'm encountering an issue while trying to deploy this contract on the zkSync sepholia Testnet.

```

Error: "underlying network changed (event=\"changed\", network={\"chainId\":324,\"name\":\"zkSync Era\"}, detectedNetwork={\"name\":\"sepolia\",\"chainId\":11155111,\"ensAddress\":null}, code=NETWORK_ERROR, version=providers/5.7.2)"

```

![image](https://github.com/user-attachments/assets/b424666d-1bc3-4093-a37c-c17976751207)

**Reply by EngrPips** [reply]
Can this be a server issue? It worked for us some hours ago.

**Reply by DCVglobalnetwork** [reply]
![image](https://github.com/user-attachments/assets/6c17e060-d3f6-46db-8a91-a7a4b259ca7f)

**Reply by DCVglobalnetwork** [reply]
I deployed a few minutes ago and went through, but this could be the testnetwork network issue. 
I will keep on deploying a few more time in different time to see if the testnet is working or there is something else

**Reply by DCVglobalnetwork** [reply]
or gas fee manually and make it default for testnet work. I'm still working on it

**Reply by EngrPips** [reply]
It seems to work sometimes and not work sometimes.

**Reply by DCVglobalnetwork** [reply]
Yes, sir, it seems like it. I did a few times and went through it, always checking if the code was correct and if I copied and pasted, there was no error in typing Also, refreshing remix  help. I keep trying a few times. 
zksync is not the only one with this issue. 

![image](https://github.com/user-attachments/assets/bae69648-41c0-4159-98cd-9035168b99f3)

![image](https://github.com/user-attachments/assets/82464676-123d-4c62-8b18-867264bf12b0)

**Reply by EngrPips** [reply]
Does it mean it is a Remix problem instead of a `ZkSync` problem?

**Reply by DCVglobalnetwork** [reply]
unsure what exactly is causing this but I found others who did try to figure out as well.

[https://github.com/NethermindEth/zksync-remix-plugin/issues/124](url)

**Reply by DCVglobalnetwork** [reply]
also could help uninstall and install again zksync plugin in remix

**Reply by DCVglobalnetwork** [reply]
again went through 
1. Refresh your remix
2. Uninstall and install the plugin again 
This is what I also figured by myself:  in some cases, if really did not work, I compiled on remote and before deploying, I connected to my wallet and went through without any issues 

![image](https://github.com/user-attachments/assets/72c0adaa-b795-414e-a3ad-41163af3aa58)

![image](https://github.com/user-attachments/assets/64d5ade0-f7b8-48b9-acf1-18fa1d1f23bd)

**Reply by C0dewithLokesh** [reply]
Thank you for the help, guys! I switched from Brave to Chrome, and it's working perfectly now. I think the browser was causing the issue."

<img width="1440" alt="image" src="https://github.com/user-attachments/assets/1466a71f-730a-4d74-8981-cd18ea3edbd0">

**Reply by EngrPips** [reply]
Awesome, Glad it finally worked for you.

**Reply by coder-002** [reply]
> also could help uninstall and install again zksync plugin in remix

Thank you @DCVglobalnetwork it worked for me
